### PR TITLE
Allow manual updates

### PR DESCRIPTION
**Pull request by garrettmflynn**
This PR makes the `TQDMProgressSubscriber` class compatible with manual updates to the `tqdm` class, as used in https://github.com/hdmf-dev/hdmf/pull/1110.

### CODE REVIEW

File: src/tqdm_publisher/_subscriber.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
-from typing import Any, Dict, Iterable
+from typing import Any, Dict, Iterable, Optional
 
 from ._publisher import TQDMProgressPublisher
 
 
 class TQDMProgressSubscriber(TQDMProgressPublisher):
-    def __init__(self, iterable: Iterable[Any], on_progress_update: callable, **tqdm_kwargs):
+    def __init__(self, on_progress_update: callable, iterable: Optional[Iterable[Any]] = None, **tqdm_kwargs):
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_subscriber.py]: To not break back compatability based on argument ordering going forward, maybe best to enforce keyword only usage here?

```suggestion
    def __init__(self, iterable: Optional[Iterable[Any]] = None, *, on_progress_update: callable, **tqdm_kwargs):
```

**Comment by CodyCBakerPhD** [comment]
Since this is the first PR after initial release, should probably include a CHANGELOG entry

File: src/tqdm_publisher/_subscriber.py (hunk 1 above)
Comment (garrettmflynn) [reply src/tqdm_publisher/_subscriber.py]: Actually, the hope would be to mirror standard `tqdm` usage as much as possible. In that case, we should enforce that `on_progress_update` is a required kwarg while allowing the user to pass anything else as they'd like.

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_subscriber.py]: @garrettmflynn OK, edited suggestion for that

Comment (garrettmflynn) [reply src/tqdm_publisher/_subscriber.py]: @CodyCBakerPhD How does this look? https://github.com/catalystneuro/tqdm_publisher/pull/65/commits/3c85e514245fb3642b99266654b179e18a42fff9

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_subscriber.py]: Fine by me

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## Unreleased
+* Update the arguments for all `tqdm_publisher` classes to mirror the `tqdm` constructor, adding additional parameters as required keyword arguments.
```
Comment (CodyCBakerPhD) [inline CHANGELOG.md]:
```suggestion
* Update the arguments for all `tqdm_publisher` classes to mirror the `tqdm` constructor, adding additional parameters as required keyword arguments. [PR #65](https://github.com/catalystneuro/tqdm_publisher/pull/65)
```

Usually include this for easier lookup/reference

File: src/tqdm_publisher/_handler.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def on_progress_update(progress_update: dict):
             """
             self.announce(message=dict(**progress_update, **additional_metadata))
 
-        return TQDMProgressSubscriber(iterable=iterable, on_progress_update=on_progress_update, **tqdm_kwargs)
+        return TQDMProgressSubscriber(*tqdm_args, on_progress_update=on_progress_update, **tqdm_kwargs)
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_handler.py:34]:
```suggestion
        return TQDMProgressSubscriber(*tqdm_args, *, on_progress_update=on_progress_update, **tqdm_kwargs)
```

Actually, to match the statement in the CHANGELOG (required), need this star here

Comment (garrettmflynn) [reply src/tqdm_publisher/_handler.py:34]: This does not seem acceptable when instantiating a class. Is this where you meant to put this?

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_handler.py:34]: No, meant in the init

Comment (garrettmflynn) [reply src/tqdm_publisher/_handler.py:34]: This is already handled by [the `*tqdm_args` argument](https://github.com/catalystneuro/tqdm_publisher/blob/f6e9a8fcb83e42c7d7ac64e64ef7f021bf8bcea0/src/tqdm_publisher/_subscriber.py#L7), right?

File: src/tqdm_publisher/_handler.py [hunk 4]
```diff
@@ -17,7 +17,7 @@ def listen(self) -> queue.Queue:
         return new_queue
 
     def create_progress_subscriber(
-        self, iterable: Iterable[Any], additional_metadata: dict = dict(), **tqdm_kwargs
+        self, *tqdm_args, additional_metadata: dict = dict(), **tqdm_kwargs
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_handler.py:20]:
```suggestion
        self, *tqdm_args, *, additional_metadata: dict = dict(), **tqdm_kwargs
```

File: src/tqdm_publisher/_subscriber.py [hunk 5]
```diff
@@ -1,11 +1,12 @@
-from typing import Any, Dict, Iterable
+from typing import Any, Dict, Iterable, Optional
 
 from ._publisher import TQDMProgressPublisher
 
 
 class TQDMProgressSubscriber(TQDMProgressPublisher):
-    def __init__(self, iterable: Iterable[Any], on_progress_update: callable, **tqdm_kwargs):
-        super().__init__(iterable=iterable, **tqdm_kwargs)
+    def __init__(self, *tqdm_args, on_progress_update: callable, **tqdm_kwargs):
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_subscriber.py:7]:
```suggestion
    def __init__(self, *tqdm_args, *, on_progress_update: callable, **tqdm_kwargs):
```

File: src/tqdm_publisher/_handler.py (hunk 3 above)
Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_handler.py:34]: No, `*` says 'everything to the right of this position is required to be a keyword argument only'

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_handler.py:34]: `*` by itself that is, `*args` just unpacks a list of ordered unnamed positional values to the signature

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_handler.py:34]: Actually I guess you're right, if it sees `*args` it imposes keyword only to the right of that

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**